docs/reference/search/search-your-data/knn-search.asciidoc

```diff
@@ -242,6 +242,117 @@ POST byte-image-index/_search
 // TEST[s/"k": 10/"k": 3/]
 // TEST[s/"num_candidates": 100/"num_candidates": 3/]
 
+[discrete]
+[[knn-search-quantized-example]]
+==== Byte quantized kNN search
+
+If you want to provide `float` vectors, but want the memory savings of `byte` vectors, you can use the
+<<dense-vector-quantization, quantization>> feature. Quantization allows you to provide `float` vectors, but
+internally they are indexed as `byte` vectors. Additionally, the original `float` vectors are still retained
+in the index. This allows you to use the original `float` vectors for re-scoring, but the `byte` vectors for
+indexing.
+
+To use quantization, you must provide a `quantization_options` object in the `dense_vector` mapping.
+
+[source,console]
+----
+PUT quantized-image-index
+{
+  "mappings": {
+    "properties": {
+      "image-vector": {
+        "type": "dense_vector",
+        "element_type": "float",
+        "dims": 2,
+        "index": true,
+        "index_options": {
+          "type": "hnsw",
+          "quantization_options": {
+            "type": "byte"
+          }
+        }
+      },
+      "title": {
+        "type": "text"
+      }
+    }
+  }
+}
+----
+// TEST[continued]
+
+. Index your `float` vectors.
++
+[source,console]
+----
+POST quantized-image-index/_bulk?refresh=true
+{ "index": { "_id": "1" } }
+{ "image-vector": [0.1, -2], "title": "moose family" }
+{ "index": { "_id": "2" } }
+{ "image-vector": [0.75, -1], "title": "alpine lake" }
+{ "index": { "_id": "3" } }
+{ "image-vector": [1.2, 0.1], "title": "full moon" }
+----
+//TEST[continued]
+
+. Run the search using the <<search-api-knn, `knn` option>>. When searching, the `float` vector is
+automatically quantized to a `byte` vector.
++
+[source,console]
+----
+POST quantized-image-index/_search
+{
+  "knn": {
+    "field": "image-vector",
+    "query_vector": [0.1, -2],
+    "k": 10,
+    "num_candidates": 100
+  },
+  "fields": [ "title" ]
+}
+----
+// TEST[continued]
+// TEST[s/"k": 10/"k": 3/]
+// TEST[s/"num_candidates": 100/"num_candidates": 3/]
+
+Since the original `float` vectors are still retained in the index, you can optionally use them for re-scoring. This will
+do the heavy query against the indexed vectors, and then you can get the absolute nearest neighbors by re-scoring.
```

Comment:
"heavy query" implies no memory footprint improvements, right? The original float vectors are loaded.

Reply:
Docs here are unclear obviously. I mean to say that the expensive query is done via approximate search against smaller foot print.